mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h

```diff
@@ -911,6 +928,35 @@ class PopulateParamsXDL : public PopulateParamsBase {
     if ((param.gemmNPerBlock % param.gemmNPerWave) != 0)
       return failure();
 
+    if ((param.gemmKPerBlock % param.gemmKPack) != 0)
+      return failure();
+
+    // Reject too wide KPACK values for fp32/fp16/bf16 types.
+    auto dataType = ctx.getDataType();
+    if (dataType.isF32() && param.gemmKPack >= 8) {
+      return failure();
+    } else if ((dataType.isF16() || dataType.isBF16()) && param.gemmKPack > 8) {
+      return failure();
+    }
+
+    // XXX FIXME: Ignore KReduction XDLOPS path for forward convolution now.
+    // These M/NPerBlock combinations will result in lowering errors at tuning.
+    if (ctx.getOpType() == miopen::ConvOpType::Fwd) {
+      if ((param.gemmMPerBlock == 16 || param.gemmMPerBlock == 32 ||
+           param.gemmMPerBlock == 64) &&
+          (param.gemmNPerBlock == 16 || param.gemmNPerBlock == 32 ||
+           param.gemmNPerBlock == 64)) {
+        return failure();
```

Comment:
Are these perhaps victims of https://ontrack-internal.amd.com/browse/SWDEV-318900

Reply:
this deserves further investigation. Currently it doesn't appear to be the case.